### PR TITLE
C3: pass arguments to underlying CLIs

### LINKED ISSUE #3797: 🐛 BUG: passing arguments into the into the framework cli is buggy
### Which Cloudflare product(s) does this pertain to?

C3

### What version(s) of the tool(s) are you using?

2.1.1

### What version of Node are you using?

18

### What operating system are you using?

Mac

### Describe the Bug

```
$ npm create cloudflare@latest -- --framework=angular -- --defaults

using create-cloudflare version 2.1.1

╭ Create an application with Cloudflare Step 1 of 3
│ 
├ In which directory do you want to create your application?
│ dir ./--defaults
│
├ What type of application do you want to create?
│ type Website or web app
│
├ Which development framework do you want to use?
│ framework Angular
│
╰ Continue with Angular via `npx @angular/cli@16.1.x new --defaults --standalone`

? What name would you like to use for the new workspace and initial project? 
```

The expected result is that C3 prompts me for directory name (and doesn't use `/--defaults` as the name), and then passes the name to the Angular CLI.

The actual result is that I'm not prompted for the name, and Angular CLI doesn't receive the directory name so it prompts me instead.

Note: using npx instead of npm fails in the same way (but requires one to not use the first `--` argument which is expected):

```
$ npx create-cloudflare@latest --framework=angular -- --defaults 

using create-cloudflare version 2.1.1

╭ Create an application with Cloudflare Step 1 of 3
│ 
├ In which directory do you want to create your application?
│ dir ./--defaults
│
├ What type of application do you want to create?
│ type Website or web app
│
├ Which development framework do you want to use?
│ framework Angular
│
╰ Continue with Angular via `npx @angular/cli@16.1.x new --defaults --standalone`

? What name would you like to use for the new workspace and initial project? 
```

### Please provide a link to a minimal reproduction

_No response_

### Please provide any relevant error logs

_No response_

### PR DESCRIPTION
**Pull request by dario-piotrowicz**
Fixes # [insert GH or internal issue number(s)].

**What this PR solves / how to test:**

**Associated docs issue(s)/PR(s):**

- [insert associated docs issue(s)/PR(s)]

**Author has included the following, where applicable:**

- [ ] Tests
- [x] Changeset ([Changeset guidelines](https://github.com/cloudflare/workers-sdk/blob/main/CONTRIBUTING.md#changesets))

**Reviewer is to perform the following, as applicable:**

- Checked for inclusion of relevant tests
- Checked for inclusion of a relevant changeset
- Checked for creation of associated docs updates
- Manually pulled down the changes and spot-tested

**Note for PR author:**

We want to celebrate and highlight awesome PR review! If you think this PR received a particularly high-caliber review, please assign it the label `highlight pr review` so future reviewers can take inspiration and learn from it.

___

resolves #3797

### CODE REVIEW

File: packages/create-cloudflare/src/cli.ts [hunk 1]
```diff
@@ -83,8 +83,12 @@ const printBanner = () => {
 	startSection(`Create an application with Cloudflare`, "Step 1 of 3");
 };
 
-export const parseArgs = async (argv: string[]): Promise<Partial<C3Args>> => {
-	const args = await yargs(hideBin(argv))
+const parseArgs = async (argv: string[]): Promise<Partial<C3Args>> => {
+	const doubleDashesIdx = argv.indexOf('--');
+	const c3Args = doubleDashesIdx < 0 ? argv : argv.slice(0, doubleDashesIdx < 0 ? undefined : doubleDashesIdx);
```
Comment (Skye-31) [inline packages/create-cloudflare/src/cli.ts:88]: This is a little redundant, either
```suggestion
	const c3Args = doubleDashesIdx < 0 ? argv : argv.slice(0, doubleDashesIdx);
```
or
```suggestion
	const c3Args = argv.slice(0, doubleDashesIdx < 0 ? undefined : doubleDashesIdx);
```
would do the same thing

Comment (dario-piotrowicz) [reply packages/create-cloudflare/src/cli.ts:88]: yeah.... refactoring leftover 😓 

Thanks! 🙂👍

**Review by jculvey — APPROVED** [review]

**Comment by dario-piotrowicz** [comment]
@jculvey I made a bunch of changes after you've approved, sorry about that 🙏 

please have another look 🙏

**Review by jculvey — APPROVED** [review]

**Comment by jculvey** [comment]
> @jculvey I made a bunch of changes after you've approved, sorry about that 🙏
> 
> please have another look 🙏

It continues to lgtm 😄 

Sorry about the merge conflicts 😅

**Pull request merged**